### DISCUSSION TITLE
Multiple CSV files  for different time instances to impose boundary conditions

### DISCUSSION THREAD

**Discussion by deepakbiitk** [idaholab/moose#25953] [Q&A General] [answered]
Dear MOOSE users,

I am solving heat equation in which I have measured temperature data at  few spatial locations on a  boundary for different times in a csv format. Multiple csv files i.e., a single measurement have a csv file. The geometry is not structured.  Is there a way I could read multiple csv files to impose boundary conditions? The boundary conditions for times between two csv, I plan to use linear interpolation.  I tried to figure out using the MOOSE site but no success yet. I appreciate your suggestions.

Best,
Deepak

**Comment by GiudGiud** [comment]
Hello

You ll want to use one CSV file per function, and use one function per boundary.
This is for a heat flux or a fixed temperature value?

Guillaume

**Reply by deepakbiitk** [reply]
Hi Guillaume,

Its fixed temperature problem. In a particular boundary, I have around 15 points where I have measured temperatures. I was thinking of using Piecewaise_multilinear, but it seems it requires a structured data. 

Deepak

**Reply by GiudGiud** [reply]
Use PiecewiseConstantFromCSV.
There's a few options for the data input but I suspect voronoi is the one to use here

**Reply by deepakbiitk** [reply]
Thanks for the suggestion. Does it also work if we have time as variable of the function, f(x,y,z,t)? Let me look into it again.

**Reply by GiudGiud** [reply]
I m afraid not. We have not coded in time dependence yet. 
For structured data there is something I think in Piecewisemultidimensial

**Reply by GiudGiud** [reply, marked as answer]
I could see a change in the PropertyReadFile to enable reading time dependence. There would be a little bit of coding work involved. 
I would recommend maybe making a derived class instead tht reads the file differently 
https://mooseframework.inl.gov/source/userobjects/PropertyReadFile.html

**Reply by caolwec** [reply]
> I could see a change in the PropertyReadFile to enable reading time dependence. There would be a little bit of coding work involved. I would recommend maybe making a derived class instead tht reads the file differently https://mooseframework.inl.gov/source/userobjects/PropertyReadFile.html

May I suggest a change to the PropertyReadFile for more convenience? - The 'prop_file_name' is for a list of property file names now, it would be better in case there are many property files if it allows a CSV file where all the property files names are listed in a column.

**Reply by GiudGiud** [reply]
That's a good idea. Could you please create an issue for the feature request?

**Reply by caolwec** [reply]
Absolutely! Thank you.

**Comment by GiudGiud** [comment]
@deepakbiitk I dont know how you made it work but I am adding what you need in this pull request
https://github.com/idaholab/moose/pull/26395

**Reply by deepakbiitk** [reply]
Hi Guillaume,

Thanks for setting this up. I am relooking to solve the problem again in MOOSE. 

Thanks,
Deepak
